name-registry/tests/functions/identity.rs

```diff
@@ -0,0 +1,46 @@
+mod success {
+    use crate::utils::{
+        abi::{identity, register, set_identity},
+        get_contract_instance,
+    };
+    use fuels::prelude::*;
+
+    #[tokio::test]
+    async fn can_get_identity() {
+        let (instance, _id, wallet, _wallet2) = get_contract_instance().await;
+
+        let wallet_identity = Identity::Address(Address::from(wallet.address()));
+
+        let name = String::from("SwaySway");
+
+        register(&instance, &name, 5000, &wallet_identity, &wallet_identity).await;
+
+        let previous_identity = identity(&instance, &name).await;
+        let wallet_identity = Identity::Address(Address::from(wallet.address()));
+
+        assert_eq!(previous_identity.0.value, wallet_identity);
+
+        let wallet2 = WalletUnlocked::new_random(None);
+        let wallet_identity2 = Identity::Address(Address::from(wallet2.address()));
+
+        set_identity(&instance, &name, wallet_identity2.clone()).await;
+
+        let new_identity = identity(&instance, &name).await;
+
+        assert_eq!(new_identity.0.value, wallet_identity2);
+    }
+}
+
+mod revert {
+    use crate::utils::{abi::identity, get_contract_instance};
+
+    #[tokio::test]
+    #[should_panic(expected = "Revert(42)")]
+    async fn cant_get_identity() {
+        let (instance, _id, _wallet, _wallet2) = get_contract_instance().await;
+
+        let name = String::from("SwaySway");
+
+        let _expiry = identity(&instance, &name).await;
```

Comment:
Instead of assigning to a variable I would `.unwrap()` and let it crash that way